### PR TITLE
Feature/login

### PR DESCRIPTION
**Pull request by TimvdLippe**

### CODE REVIEW

File: src/main/java/nl/tudelft/planningstool/api/security/AuthorizationFilter.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        } catch (Exception e) {
+            requestContext.abortWith(
+                    Response.status(Response.Status.FORBIDDEN).build());
+        }
+    }
+
+    // Extract the roles from the annotated element
+    private List<Role> extractRoles(AnnotatedElement annotatedElement) {
+        if (annotatedElement == null) {
+            return new ArrayList<Role>();
```
Comment (jwgmeligmeyling) [inline src/main/java/nl/tudelft/planningstool/api/security/AuthorizationFilter.java]: `Collections.emptyList()` is cheaper to instantiate (and in the JDK).

**Comment by jwgmeligmeyling** [comment]
Implementation of the `AuthenticationFilter` looks good! Seems the added dependency to `jetty-security` is still unused, are you planning to use it in the future? As it seems you're planning to do token based authentication (by query parameter?) I don't think you actually need it. Keep in mind that when doing token based authentication users should not be allowed to exchange urls or they will give away their identity (if it's internal API or this constraint is very clear there is no problem). Also be sure that the token generation is relatively strong (properly seeded rng)

**Comment by TimvdLippe** [comment]
The tokens are only user in client-server communication, so the users should not see anything directly. The dependency was added by Gijs, he knows if we need it later or not.

I think the commented code was the generator code, is that secure enough?

**Comment by jwgmeligmeyling** [comment]
Ah I missed that code. SecureRandom as basis for a (relatively long) string should be enough!

**Comment by GijsWeterings** [comment]
`jetty-security` is leftover from first implementation. I'll nuke that :shipit:

**Comment by jwgmeligmeyling** [comment]
:+1:

**Pull request merged**